src/TriggerBinding/SqlTriggerValueProvider.cs

```diff
@@ -24,21 +27,37 @@ internal class SqlTriggerValueProvider : IValueProvider
         /// <param name="tableName">Name of the user table</param>
         public SqlTriggerValueProvider(Type parameterType, object value, string tableName)
         {
-            this.Type = parameterType;
+            this._parameterType = parameterType;
             this._value = value;
             this._tableName = tableName;
+            this._isString = parameterType == typeof(string);
         }
 
         /// <summary>
         /// Gets the trigger argument value.
         /// </summary>
-        public Type Type { get; }
+        public Type Type
+        {
+            get
+            {
+                if (this._isString)
+                {
+                    return typeof(IReadOnlyCollection<JObject>);
+                }
+                return this._parameterType;
+            }
+        }
 
         /// <summary>
         /// Returns value of the trigger argument.
         /// </summary>
         public Task<object> GetValueAsync()
         {
+            if (this._isString)
+            {
+                return Task.FromResult<object>(JArray.FromObject(this._value).ToString(Newtonsoft.Json.Formatting.None));
```

Comment:
You can't use `JArray.FromObject` directly or we'll run into the issues I fixed previously with picking up global default settings.

https://github.com/Azure/azure-functions-sql-extension/pull/703

You should use the overload that takes in a serializer (and ideally put this with the other helpers in Utils.cs) : https://github.com/JamesNK/Newtonsoft.Json/blob/master/Src/Newtonsoft.Json/Linq/JArray.cs#L202

Reply:
do we have tests for this for input/output bindings already? (i.e. non-default global settings)

Let's create an issue either for this to create tests to ensure we don't regress this scenario -- either just for trigger (if tests for bindings already exist), or for both.

Reply:
I added https://github.com/Azure/azure-functions-sql-extension/blob/main/test/Integration/test-csharp/Startup.cs - so we should be getting validation of that. But that only covers the functions in the test folder so we might need/want to expand it to be applied to our "sample" functions as well (but only during testing) if we find that the test folder coverage isn't enough.

Reply:
Now that I think about it - I believe technically this doesn't need to follow the same logic since this code should only be hit for non-C# languages. But those can't override the default Newtonsoft settings - so we should be safe. But I also don't think it hurts to do this and might save us in case there's an edge case where it can happen (for example, I suppose it's possible another extension changes the default settings....).

Reply:
Updated to use Utils.JsonSerializeObject instead of converting to JArray and using ToString(). I tested the different languages and the results were the same. This way the conversion from JObject to JSON string is consistent between input binding and trigger.